pkg/csi/manila/util/exportlocation.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/gophercloud/gophercloud/openstack/sharedfilesystems/v2/shares"
+)
+
+// Predicate type for filtering out export locations.
+// It's supplied with the index into the export location slice that's being scanned.
+// Should return (true, nil) when search conditions are satisfied,
+// and (false, nil) when they are not.
+type ExportLocationPredicate func(index int) (match bool, err error)
+
+// Predicate matching any export location
+func AnyExportLocation(int) (bool, error) { return true, nil }
```

Comment:
Are any other predicates supported now?

Reply:
Not yet, we're planning two more for now: 1) exact match, and 2) subnet match.